pkg/updater/org_config.go

```diff
@@ -62,12 +78,58 @@ func (c *OrgConfig) GetPackage(_ context.Context, pkg string, version string) (P
 
 // GetDefaultPackage returns the default version for the given package.
 // The function blocks until the catalog and org preferences are received from RC.
-// TODO: Implement with RC support instead of hardcoded files.
-func (c *OrgConfig) GetDefaultPackage(ctx context.Context, pkg string) (Package, error) {
+// TODO: Implement with RC support instead of hardcoded default file.
+func (c *orgConfig) GetDefaultPackage(ctx context.Context, pkg string) (Package, error) {
 	var defaults map[string]string
 	err := json.Unmarshal(rawDefaults, &defaults)
 	if err != nil {
 		return Package{}, fmt.Errorf("could not unmarshal defaults: %w", err)
 	}
 	return c.GetPackage(ctx, pkg, defaults[pkg])
 }
+
+func (c *orgConfig) waitForCatalog(ctx context.Context) error {
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case <-c.catalogReceived:
+		return nil
+	}
+}
+
+func (c *orgConfig) onCatalogUpdate(catalogConfigs map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	var mergedCatalog catalog
+	for configPath, config := range catalogConfigs {
+		var catalog catalog
+		err := json.Unmarshal(config.Config, &catalog)
+		if err != nil {
+			log.Errorf("could not unmarshal updater catalog: %s", err)
+			applyStateCallback(configPath, state.ApplyStatus{State: state.ApplyStateError, Error: err.Error()})
+			return
+		}
+		mergedCatalog.Packages = append(mergedCatalog.Packages, catalog.Packages...)
+	}
+	for configPath := range catalogConfigs {
+		applyStateCallback(configPath, state.ApplyStatus{State: state.ApplyStateAcknowledged})
+	}
```

Comment:
Would it make sense to mark each succesful `configPath` in the loop above and avoid a second iteration over `catalogConfigs`?

Reply:
The above loop can exit anytime on an error so I think it's cleaner if the reported state is `unacked` + `error` instead of a mix of `unacked` / `acked` depending on ordering + `error`